source/class/qx/tool/cli/commands/Deploy.js

```diff
@@ -0,0 +1,158 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2020 Zenesis Ltd https://www.zenesis.com
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * John Spackman (john.spackman@zenesis.com, @johnspackman)
+
+************************************************************************ */
+require("./Compile");
+
+require("@qooxdoo/framework");
+const fs = require("fs");
+const path = require("upath");
+const stream = require("stream");
+const util = require("util");
+const pipeline = util.promisify(stream.pipeline);
+
+/**
+ * Build and deploy a project
+ * 
+ */
+qx.Class.define("qx.tool.cli.commands.Deploy", {
+  extend: qx.tool.cli.commands.Command,
+  statics: {
+    getYargsCommand: function() {
+      return {
+        command: "deploy [options]",
+        describe: "deploys qooxdoo application(s)",
+        builder: {
+          "out": {
+            describe: "Output directory for the deployment",
+            demandOption: true
+          },
+          "app-name": {
+            describe: "The name of the application to deploy (default is all apps)",
+            nargs: 1,
+            type: "string"
+          },
+          "source-maps": {
+            describe: "Enable source maps",
+            type: "boolean",
+            default: null
+          },
+          "clean": {
+            describe: "Deletes the application output directory before deploying",
+            type: "boolean",
+            default: false
+          }
+        }
+      };
+    }
+  },
+
+  members: {
+
+    /*
+     * @Override
+     */
+    process: async function() {
+      let argv = this.argv;
+      
+      let config = this.__config = await qx.tool.cli.Cli.getInstance().getParsedArgs();
+      if (!config) {
+        throw new qx.tool.utils.Utils.UserError("Error: Cannot find any configuration");
+      }
+      
+      if (argv.sourceMaps === null) {
+        qx.tool.compiler.Console.print("qx.tool.cli.deploy.sourceMapsNotSpecified");
+      }
+      
+      if (!argv.clean) {
+        qx.tool.compiler.Console.print("qx.tool.cli.deploy.notClean");
+      }
+      
+      config.targetType = "build";
```

Comment:
This should probably be made configurable (later) so that one can supply the name of a custom build target